platform/src/main/python/dlpx/virtualization/platform/_upgrade.py

```diff
@@ -95,28 +113,26 @@ def _success_upgrade_response(upgraded_dict):
             return_value=upgrade_result)
         return upgrade_response
 
-    def __process_upgrade_request(self, request, id_to_impl):
-        """Iterate through all objects in the pre_upgrade_parameters map,
-        invoke all available migrations on each object and its metadata,
-        and return a map containing the updated metadata for each object.
+    @staticmethod
+    def _run_migration_upgrades(request, lua_impls_getter,
+                                platform_impls_getter):
+        """
+        Given the list of lua and platform migration to run, iterate and
+        invoke these migrations on each object and its metadata, and return a
+        dict containing the upgraded parameters.
         """
         post_upgrade_parameters = {}
+        impls_list = lua_impls_getter(
+            request.lua_upgrade_version) + platform_impls_getter(
+                request.migration_ids)
         for (object_ref, metadata) in request.pre_upgrade_parameters.items():
             # Load the object metadata into a dictionary
             current_metadata = json.loads(metadata)
-            #
-            # Loop through all migrations that were passed into the upgrade
-            # request. Protobuf will preserve the ordering of repeated
-            # elements, so we can rely on the backend to sort the migration
```

Comment:
Might make sense to keep some of this information around regarding protobuf preserving the ordering of repeated elements.

Reply:
👍 moved it around to leave a comment for the `platform_impls_getter(request.migration_ids)` call instead. lemme know if the comment makes sense or should be moved.